src/js/msp/MSPHelper.js

```diff
@@ -2615,15 +2617,15 @@ MspHelper.prototype.sendLedStripConfig = function(onCompleteCallback) {
             mask |= (led.y << 0);
             mask |= (led.x << 4);
 
-            for (const functionLetterIndex of led.functions) {
+            for (const functionLetterIndex in led.functions) {
```

Comment:
Do we want to include all non-array properties of `led.functions` in this? If not then this is wrong, and the line below should be changed.

Reply:
Changed back to standard for loop is much more clear.

Reply:
Using `for ( of )` would be perfectly fine in these cases:

```
            for (const directionLetter of led.directions) {
￼                const bitIndex = ledDirectionLetters.indexOf(directionLetter);
￼                if (bitIndex >= 0) {
￼                    directionMask = bit_set(directionMask, bitIndex);
                }
            }
```

However, `for ( in )` is not ok, as it iterates over a different set.

Reply:
Somehow for ( of ) didn't work here, and for ( in ) did, but it's recommended not to use it as index while iterating over an array:
https://stackoverflow.com/questions/500504/why-is-using-for-in-for-array-iteration-a-bad-idea